src/server/FileUploadDataSource.ts

```diff
@@ -0,0 +1,265 @@
+import {
+  GraphQLDataSourceProcessOptions,
+  RemoteGraphQLDataSource,
+} from '@apollo/gateway';
+import { GraphQLResponse } from 'apollo-server-types';
+import { FileUpload, Upload } from 'graphql-upload';
+import { Headers, Response } from 'apollo-server-env';
+import { isObject } from '@apollo/gateway/dist/utilities/predicates';
+
+import { cloneDeep, set } from 'lodash';
+import { Request } from 'node-fetch';
+import FormData from './FormData';
+import ProfusionFileUploadDataSource from '@profusion/apollo-federation-upload';
+
+/**
+ * These types and functions below are copied from the library as well.
+ * Since they're not class members and aren't exported, I had to copy them
+ */
```

Comment:
Could you add a note here which library the types & functions are copied from right here instead of, or in addition to further down the file?